bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/i18n/I18nProviderImpl.java

```diff
@@ -90,13 +90,13 @@ protected void deactivate(ComponentContext componentContext) {
     }
 
     @Modified
-    protected synchronized void modified(Map<String, Object> config) {
-        final String language = (String) config.get(LANGUAGE);
-        final String script = (String) config.get(SCRIPT);
-        final String region = (String) config.get(REGION);
-        final String variant = (String) config.get(VARIANT);
-        final String location = (String) config.get(LOCATION);
-        final ZoneId timeZone = (ZoneId) config.get(TIMEZONE);
+    protected synchronized void modified(Map<String, String> config) {
+        final String language = config.get(LANGUAGE);
+        final String script = config.get(SCRIPT);
+        final String region = config.get(REGION);
+        final String variant = config.get(VARIANT);
+        final String location = config.get(LOCATION);
+        final ZoneId timeZone = StringUtils.isBlank(config.get(TIMEZONE)) ? null : ZoneId.of(config.get(TIMEZONE));
```

Comment:
Wouldn't it be cleaner to turn this variable into a string and make `setTimeZone(...)` accept a string accordingly. Afair `TimeZone.getTimeZone(...)` has an overloaded method which accepts a string directly. Or would the result be any different?

Reply:
agreed. Looks even better to pass a String to `TimeZone.getTimeZone` instead of a ZoneId and then retrieving the ZoneId....